# HG changeset patch
# User Rik <rik@octave.org>
# Date 1542308607 28800
#      Thu Nov 15 11:03:27 2018 -0800
# Node ID 421ea6654fa469226ab6efc82a242aadd62e54bf
# Parent  5d28fc32a7c77eddc3071fcdf8f0d033cf226243
strsplit.m: use isrow() to check for row vector input.

* strsplit.m: Validate input STR is a char row vector with isrow().

diff --git a/scripts/strings/strsplit.m b/scripts/strings/strsplit.m
--- a/scripts/strings/strsplit.m
+++ b/scripts/strings/strsplit.m
@@ -172,17 +172,17 @@ function [cstr, matches] = strsplit (str
       del = {" ", "\f", "\n", "\r", "\t", "\v"};
     else
       del = '\s';
     endif
   endif
 
   if (! ischar (str) || (! ischar (del) && ! iscellstr (del)))
     error ("strsplit: S and DEL must be string values");
-  elseif (numel (str) != columns (str))
+  elseif (! isempty (str) && ! isrow (str))
     error ("strsplit: S must be a char row vector")
   elseif (! isscalar (args.collapsedelimiters))
     error ("strsplit: COLLAPSEDELIMITERS must be a scalar value");
   endif
 
   if (strncmpi (args.delimitertype, "simple", length_deltype))
     is_simple = true;
   elseif (strncmpi (args.delimitertype, "regularexpression", length_deltype))
